### PR TITLE
CPDRP-1000 Participant eligibility logic update

### PR DESCRIPTION
**Pull request by tonyheadford**
## Ticket and context

Ticket: [Jira](https://dfedigital.atlassian.net/browse/CPDRP-1000)
Updates participant eligibility logic to account for the participant profile type.
Adds reason enum so that we can easily see the reason for the given status.

We will want to run this manually against prod, primarily to update the status for mentors that have a previous induction. Currently these are flagged for manual check but we should ignore the previous induction flag for mentors.

```ruby
ECFParticipantEligibility.where(manually_validated: false).find_each { |eligibility| eligibility.determine_status; eligibility.save! }
```

### Code quality checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### Gotchas
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.

## External API changes

Does this change anything in our external APIs? If so, did you remember to update the CHANGELOG for Lead Providers? (lead-providers/guidance/release-notes)

### CODE REVIEW

File: db/migrate/20210930125828_add_reason_to_ecf_participant_eligibilities.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddReasonToECFParticipantEligibilities < ActiveRecord::Migration[6.1]
+  def change
+    add_column :ecf_participant_eligibilities, :reason, :string, null: false, default: "none"
```
Comment (kicferk1) [inline db/migrate/20210930125828_add_reason_to_ecf_participant_eligibilities.rb:5]: Is this reason for the participant not being eligible?

If so, can we call it `ineligible_reason`? `reason_ineligible`?

Comment (jamesrose) [reply db/migrate/20210930125828_add_reason_to_ecf_participant_eligibilities.rb:5]: Not quite. I think it covers any status reason.

Comment (tonyheadford) [reply db/migrate/20210930125828_add_reason_to_ecf_participant_eligibilities.rb:5]: I guess it's the reason for the status, there aren't any reasons for `eligible` but you'd have to put something in hence the `none` reason.

**Comment by ethanmills** [comment]
I think it would be safer to run 

```ruby
ECFParticipantEligibility.where(manually_validated: false).find_each { |eligibility| eligibility.determine_status; eligibility.save! }
```

otherwise we risk reversing any manual override to eligible

**Comment by tonyheadford** [comment]
> I think it would be safer to run
> 
> ```ruby
> ECFParticipantEligibility.where(manually_validated: false).find_each { |eligibility| eligibility.determine_status; eligibility.save! }
> ```
> 
> otherwise we risk reversing any manual override to eligible

Yes, good idea (although I wasn't aware that there had been any as of yet?)

**Review by ethanmills — APPROVED** [review]

**Pull request merged**